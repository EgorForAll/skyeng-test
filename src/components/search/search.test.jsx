import React from "react";
import { render, screen } from "@testing-library/react";
import configureStore from "redux-mock-store";
import * as redux from "react-redux";
import "@testing-library/jest-dom";
import Search from "./search";

const mockStore = configureStore();

describe("SearchComponent", () => {
  it("SearchComponent schould render correctly", () => {
    render(
      <redux.Provider store={mockStore({})}>
        <Search />
      </redux.Provider>
    );
    expect(screen.getByLabelText(/По возрастанию/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/По убыванию/i)).toBeInTheDocument();
    expect(
      screen.getByText("Сортировать по количеству публичных репозиториев:")
    ).toBeInTheDocument();
    expect(screen.getByRole("button", { name: "Поиск" }));
  });
  it("SearchComponent snapshot", () => {
    const search = render(
      <redux.Provider store={mockStore({})}>
        <Search />
      </redux.Provider>
    );
    expect(search).toMatchSnapshot();
  });
});
