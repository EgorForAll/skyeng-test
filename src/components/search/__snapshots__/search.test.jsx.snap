// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent SearchComponent snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action=""
        class="form"
        method="GET"
      >
        <div
          class="search-bar"
        >
          <label
            class="visually-hidden"
            for="search"
          >
            Строка поиска
          </label>
          <input
            class="search-input"
            id="search"
            placeholder="введите логин пользователя"
            type="text"
          />
          <button
            class="submit-btn"
            type="submit"
          >
            Поиск
          </button>
        </div>
        <div
          class="radio-buttons"
        >
          <span>
            Сортировать по количеству публичных репозиториев:
          </span>
          <input
            checked=""
            id="up-radio"
            name="radio"
            type="radio"
          />
          <label
            for="up-radio"
          >
            По возрастанию
          </label>
          <input
            id="down-radio"
            name="radio"
            type="radio"
          />
          <label
            for="down-radio"
          >
            По убыванию
          </label>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action=""
      class="form"
      method="GET"
    >
      <div
        class="search-bar"
      >
        <label
          class="visually-hidden"
          for="search"
        >
          Строка поиска
        </label>
        <input
          class="search-input"
          id="search"
          placeholder="введите логин пользователя"
          type="text"
        />
        <button
          class="submit-btn"
          type="submit"
        >
          Поиск
        </button>
      </div>
      <div
        class="radio-buttons"
      >
        <span>
          Сортировать по количеству публичных репозиториев:
        </span>
        <input
          checked=""
          id="up-radio"
          name="radio"
          type="radio"
        />
        <label
          for="up-radio"
        >
          По возрастанию
        </label>
        <input
          id="down-radio"
          name="radio"
          type="radio"
        />
        <label
          for="down-radio"
        >
          По убыванию
        </label>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
